sputnikdao-factory2/src/lib.rs

```diff
@@ -109,30 +80,115 @@ impl SputnikDAOFactory {
             false
         }
     }
+
+    /// Tries to upgrade given account created by this factory to the latest code.
+    pub fn upgrade(&self, account_id: AccountId) {
+        self.assert_owner();
+        assert!(
+            self.daos.contains(&account_id),
+            "Must be contract created by factory"
+        );
+        self.factory_manager
+            .update_contract(account_id, self.get_latest_code_hash(), "update");
```

Comment:
Doesn't this call require the DAO to have an `update` method? I'm not seeing an implementation for `update` inside the DAO.

Reply:
You can ignore this comment, I've just seen you pushed a new commit with the implementation for `update`.